plugin-interface/src/main/scala/mesosphere/marathon/plugin/EnvVarValue.scala

```diff
@@ -0,0 +1,11 @@
+package mesosphere.marathon.plugin
+
+trait EnvVarValue
+
+trait EnvVarString extends EnvVarValue {
```

Comment:
maybe that's what gaston was suggesting? the plugin API looked like mostly
traits so I was trying to stick with that. also, case classes can't be
extended. so if we made this a case class here, then we'd delete the
EnvVarString case class in mesosphere/marathon/state .. does that sound OK?

On Wed, May 11, 2016 at 4:34 AM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In
> plugin-interface/src/main/scala/mesosphere/marathon/plugin/EnvVarValue.scala
> <https://github.com/mesosphere/marathon/pull/3856#discussion_r62809893>:
>
> > @@ -0,0 +1,11 @@
> > +package mesosphere.marathon.plugin
> > +
> > +trait EnvVarValue
> > +
> > +trait EnvVarString extends EnvVarValue {
>
> Why no case class? That would give you static case checking.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/marathon/pull/3856/files/e24d8fc72711094a53f54eb5aef6d9d2fb073b52#r62809893>
>